src/python-common/ceph/deployment/service_spec.py

```diff
@@ -680,7 +680,9 @@ def validate(self) -> None:
                         f'Service of type \'{self.service_type}\' should not contain a service id')
 
         if self.service_id:
-            if not re.match('^[a-zA-Z0-9_.-]+$', self.service_id):
+            if self.service_type == 'mds' and str(self.service_id)[0].isdigit():
+                raise SpecValidationError('MDS service id cannot start with a numeric digit')
+            elif not re.match('^[a-zA-Z0-9_.-]+$', self.service_id): 
```

Comment:
should we consider having a well-defined `MDSSpec` class specifically for this new validation?

Reply:
In my opinion current code is simple and easy to understand. In case we start adding more specific checks for special services then I'd also vote to go and create a dedicated sub-class for each service. I think we should also consider for the future introducing some kind of schema (jsonschema i.e) to validate this kind of data classes structure and fields.

Reply:
there is actually precedent for putting fairly simple checks in validation functions for classes for service specs for specific services. E.g. the iscsi service spec has 

![image](https://user-images.githubusercontent.com/47704447/153094903-ad5e97fd-1e07-4cd9-bd8e-2c093fbfe057.png)

and alertmanager service spec has

![image](https://user-images.githubusercontent.com/47704447/153095309-5f9b3457-b30e-4084-bab7-06dfd62ba3b3.png)

so it's not out of the question to do something like that. While this is a fairly minor change, I think it's the only place in the generic service spec class that we're doing a special check for a certain service spec. For organizational reasons it could be worth creating an mds service spec class to be in line with how we've done it for other services. It would be super simple as well.

Reply:
I didn't notice this subclasses. Since we are already doing the same for other similar cases I agree to create a new subclass for MDS spec also. I'll change the code.